dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -1177,6 +1146,38 @@ private void processWesCommands(final List<String> args) {
 
     }
 
+    /**
+     * This will aggregate the WES request URI and credentials into a single object for use down the line
+     * @param args The commaand line arguments
+     */
+    private void aggregateWesRequestData(final List<String> args) {
+
+        // Get the config file to see if credentials are there
+        INIConfiguration config = Utilities.parseConfig(this.getConfigFile());
+        SubnodeConfiguration configSubNode = null;
+        try {
+            configSubNode = config.getSection("WES");
+        } catch (Exception e) {
+            out("Could not get WES section from config file");
+        }
+
+        // Attempt to find the WES URI
+        final String wesUri = optVal(args, "--wes-url", null);
+        final String wesEndpointUrl = ObjectUtils.firstNonNull(wesUri, Objects.requireNonNull(configSubNode).getString("url"));
+
+        // Depending on the endpoint (AWS/non-AWS) we need to look for a different set of credentials
+        final boolean isAwsWes = flagPresent(args, "--aws");
+        if (isAwsWes) {
+            // TODO should probably have the config sections all defined within a single class. At least have enums.
+            final String accessKey = ObjectUtils.firstNonNull(optVal(args, "--aws-access-key", null), Objects.requireNonNull(configSubNode).getString("aws-access-key"));
```

Comment:
Sounds reasonable to me, profiles feel like the ideal way to pass credentials. I'd suggest that as a TODO  though (unless there are strong feelings otherwise), as AWS credentials might take some time to parse correctly: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html, as we need both keys and regions...

My initial thoughts were to support `PERMANENT`, `TEMPORARY`, and `PROFILE` credentials for AWS. Currently the second 2 are future enhancements for the credentials object: https://github.com/dockstore/dockstore-cli/blob/838adf8e65ff5b826fa4803ff0409312022b3d86/dockstore-client/src/main/java/io/dockstore/client/cli/nested/WesRequestData.java#L28 If we don't want to have AWS credentials specified in a config file at all, I can remove the lines that search for it.